diff --git a/src/programs/uprobe.rs b/src/programs/uprobe.rs
index e26aaef..9d1bf31 100644
--- a/src/programs/uprobe.rs
+++ b/src/programs/uprobe.rs
@@ -1082,7 +1082,7 @@ mod tests {
         assert_matches!(
             find_debug_path_in_object(&main_obj, main_path, "symbol"),
             Ok(path) => {
-                assert_eq!(&*path, "/path/to/main.debug", "path: {}", path.display());
+                assert_eq!(&*path, Path::new("/path/to/main.debug"), "path: {}", path.display());
             }
         );
     }
@@ -1286,7 +1286,7 @@ mod tests {
         assert_matches!(
             proc_map_libs.find_library_path_by_name(Path::new("libcrypto.so.3.0.9")),
             Ok(Some(path)) => {
-                assert_eq!(path, "/usr/lib64/libcrypto.so.3.0.9", "path: {}", path.display());
+                assert_eq!(path, Path::new("/usr/lib64/libcrypto.so.3.0.9"), "path: {}", path.display());
             }
         );
     }
@@ -1303,7 +1303,7 @@ mod tests {
         assert_matches!(
             proc_map_libs.find_library_path_by_name(Path::new("libcrypto")),
             Ok(Some(path)) => {
-                assert_eq!(path, "/usr/lib64/libcrypto.so.3.0.9", "path: {}", path.display());
+                assert_eq!(path, Path::new("/usr/lib64/libcrypto.so.3.0.9"), "path: {}", path.display());
             }
         );
     }
@@ -1324,7 +1324,7 @@ mod tests {
         assert_matches!(
             proc_map_libs.find_library_path_by_name(Path::new("ld-linux-x86-64.so.2")),
             Ok(Some(path)) => {
-                assert_eq!(path, "/usr/lib64/ld-linux-x86-64.so.2", "path: {}", path.display());
+                assert_eq!(path, Path::new("/usr/lib64/ld-linux-x86-64.so.2"), "path: {}", path.display());
             }
         );
     }
